vendor/README.md

```diff
@@ -0,0 +1,319 @@
+
+
+  1 passing (1s)
+```
+
+That's looking good!
+
+Next we'll add a second test for a branch build.
+
+```js
+t.create('build status on named branch')
+  .get('/rust-lang/rust/stable.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('build'),
+    value: Joi.equal('failing', 'passing', 'unknown')
+  }));
+```
+
+```
+http://localhost:1111/try.html
+  Travis
+    build status on default branch
+      ✓
+  [ GET http://localhost:1111/travis/rust-lang/rust.json ] (220ms)
+    build status on named branch
+      ✓
+  [ GET http://localhost:1111/travis/rust-lang/rust/stable.json ] (100ms)
+
+
+  2 passing (1s)
+```
+
+Having covered the typical and customize cases, we'll move on to errors.
+
+First, a nonexistent repo, which Travis reports as having an `unknown` status:
+
+```js
+t.create('unknown repo')
+  .get('/this-repo/does-not-exist.json')
+  .expectJSON({ name: 'build', value: 'unknown' });
+```
+
+Since in this case we know the exact badge which should be returned, we can
+use the more concise `expectJSON()` in place of `expectJSONTypes()`.
+
+Next, we want to cover the code in the `catch` block. To do this, we need to
+trigger an exception. After studying the code, we realize this could happen on
+a request without a Content-Disposition header.
+
+Since we don't have an easy way to get the server to return a real repository
+request without a Content-Disposition header, we will intercept the request
+and provide our own mock response. We use the `intercept()` method provided by
+the [icedfrisby-nock plugin][icedfrisby-nock]. It takes a setup function,
+which returns an interceptor, and exposes the full API of the HTTP mocking
+library [Nock][].
+
+```js
+t.create('missing content-disposition header')
+  .get('/foo/bar.json')
+  .intercept(nock => nock('https://api.travis-ci.org')
+    .head('/foo/bar.svg')
+    .reply(200))
+  .expectJSON({ name: 'build', value: 'invalid' });
+```
+
+Nock is fussy. All parts of a request must match perfectly for the mock to
+take effect, including the method (in this case HEAD), scheme (https), host,
+and path.
+
+[icedfrisby-nock]: https://github.com/paulmelnikow/icedfrisby-nock#usage
+[Nock]: https://github.com/node-nock/nock
+
+
+Code coverage
+-------------
+
+By checking code coverage, we can make sure we've covered all our bases.
+
+We can generate a coverage report and open it:
+
+```
+npm run coverage:test:vendor -- -- --only=travis
+npm run coverage:report:open
+```
+
+Note the two sets of double dashes.
+
+After searching `server.js` for the Travis code, we see that we've missed a
+big block: the error branch in the request callback. To test that, we mock a
+connection error.
+
+```js
+t.create('connection error')
+  .get('/foo/bar.json')
+  .intercept(nock => nock('https://api.travis-ci.org')
+    .head('/foo/bar.svg')
+    .replyWithError({ code: 'ECONNRESET' }))
+  .expectJSON({ name: 'build', value: 'invalid' });
+```
+
+
+Getting help
```

Comment:
This could be at the beginning. If I needed help during the tutorial, I would not have reached this.